packages/@lwc/template-compiler/src/codegen/index.ts

```diff
@@ -244,14 +244,16 @@ function transform(codeGen: CodeGen): t.Expression {
             transformChildren(conditionalParentBlock)
         );
 
-        let elseExpression: t.Expression = t.literal(null);
+        let elseExpression: t.Expression;
         if (conditionalParentBlock.else) {
             elseExpression = isElseifBlock(conditionalParentBlock.else)
                 ? transformConditionalParentBlock(conditionalParentBlock.else, ifBlockKey)
                 : codeGen.genFragment(
                       t.literal(ifBlockKey),
                       transformChildren(conditionalParentBlock.else)
                   );
+        } else {
+            elseExpression = codeGen.genFragment(t.literal(ifBlockKey), t.arrayExpression([]));
```

Comment:
@jmsjtu this bug fix required a minor change to the codegen for the else branch. Would love your thoughts here too.